### PR TITLE
tests-invoke: Continuous collision detection

### PR DESCRIPTION
**Pull request by Gundersanne**

### CODE REVIEW

File: tests-invoke [hunk 1]
```diff
@@ -252,8 +256,21 @@ class PullTask(object):
             cmd = ["timeout", "120m", entry_point]
             if opts.verbose:
                 cmd.append("--verbose")
-            ret = subprocess.call(cmd)
-
+            # Popen instead of call; while not etc... (SubProcess.poll)
```
Comment (martinpitt) [inline tests-invoke]: Please drop this comment. Or better even, say "# cancel test if PR gets updated while test is running"

File: tests-invoke [hunk 2]
```diff
@@ -252,8 +256,21 @@ class PullTask(object):
...
+            if ret:
```
Comment (martinpitt) [inline tests-invoke]: This looks out of place -- this is the "then" branch of `if not ret:`, and it's not reassigned anywhere in between. Probably this whole if is just an overzealous copy&paste?

File: tests-invoke [hunk 3]
```diff
@@ -252,8 +256,21 @@ class PullTask(object):
...
+                sys.stderr.write('Collision detected: {0}\n'.format(ret))
+                return None
+            p = subprocess.Popen(cmd)
+            while p.poll() is None:
+                ret = self.detect_collisions(opts, True)
+                if ret:
```
Comment (martinpitt) [inline tests-invoke]: No need to use the `ret` variable here.

File: tests-invoke [hunk 4]
```diff
@@ -252,8 +256,21 @@ class PullTask(object):
...
+                    sys.stderr.write('Collision detected: {0}\n'.format(ret))
```
Comment (martinpitt) [inline tests-invoke]: Let's please make this a different message. "PR got updated, cancelling this test run."?

File: tests-invoke [hunk 5]
```diff
@@ -252,8 +256,21 @@ class PullTask(object):
...
+                    ret = None
+                    p.terminate()
```
Comment (martinpitt) [inline tests-invoke]: Do a `break` here, or do the sleeping in `else:`. No need to sleep another minute after cancelling.

Also, does the child get reaped automatically, or do we need a `p.wait()` here?

File: tests-invoke [hunk 6]
```diff
@@ -252,8 +256,21 @@ class PullTask(object):
...
+                time.sleep(60)
+            ret = p.returncode
+            # if collision
+            sys.stderr.write("RETURN CODE: {0}", p.returncode)
```
Comment (martinpitt) [inline tests-invoke]: It's useful to log the return code, but the comment doesn't fit.

File: tests-invoke (hunk 4 above)
Comment (Gundersanne) [reply tests-invoke]: Oh I chose this because it's  similar to the comment on line 225, but I'll add the fact that it's cancelling a running test?

**Pull request merged**